core/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -73,15 +73,15 @@ public MultiSearchRequest add(SearchRequest request) {
     }
 
     public MultiSearchRequest add(byte[] data, int from, int length,
-                                  @Nullable String[] indices, @Nullable String[] types, @Nullable String searchType) throws Exception {
-        return add(new BytesArray(data, from, length), indices, types, searchType, null, IndicesOptions.strictExpandOpenAndForbidClosed(), true);
+            @Nullable String path, @Nullable String[] indices, @Nullable String[] types, @Nullable String searchType) throws Exception {
```

Comment:
Maybe this parameter would be better as a boolean so the actual path name and parsing is kept in the REST layer especially as the rest of the path has already been parsed into the indices and types at this point. Also for the Java API it means that users just need a boolean saying that it's a template search instead of needing to fake a path?